src/preferences/dialog/dlgprefcolors.cpp

```diff
@@ -94,7 +118,84 @@ void DlgPrefColors::slotApply() {
                         m_colorPaletteSettings.getTrackColorPalette()));
     }
 
-    m_pConfig->setValue(
-            ConfigKey("[Controls]", "auto_hotcue_colors"),
-            checkBoxAssignHotcueColors->isChecked());
+    int index = comboBoxHotcueDefaultColor->currentIndex();
+
+    if (index > 0) {
+        m_pConfig->setValue(ConfigKey("[Controls]", "auto_hotcue_colors"), false);
+        m_pConfig->setValue(ConfigKey("[Controls]", "HotcueDefaultColorIndex"), index - 1);
+    } else {
+        m_pConfig->setValue(ConfigKey("[Controls]", "auto_hotcue_colors"), true);
+        m_pConfig->setValue(ConfigKey("[Controls]", "HotcueDefaultColorIndex"), -1);
+    }
+}
+
+QPixmap DlgPrefColors::drawPalettePreview(const QString& paletteName) {
+    foreach (const ColorPalette& palette, mixxx::PredefinedColorPalettes::kPalettes) {
+        if (paletteName == palette.getName()) {
+            int count = math_max(palette.size(), 1);
+            int width = math_min((200 / count), 16);
+            QPixmap pixmap(count * width, 16);
+            pixmap.fill(Qt::black);
+            QPainter painter(&pixmap);
+            for (int i = 0; i < palette.size(); ++i) {
+                painter.setPen(mixxx::RgbColor::toQColor(palette.at(i)));
+                painter.setBrush(mixxx::RgbColor::toQColor(palette.at(i)));
+                painter.drawRect(i * width, 0, width, 16);
+            }
+            return pixmap;
+        }
+    }
+    return QPixmap();
+}
+
+void DlgPrefColors::slotTrackPaletteChanged(const QString& paletteName) {
+    QPixmap pixmap = drawPalettePreview(paletteName);
+    labelTrackPalette->setPixmap(pixmap);
+}
+
+void DlgPrefColors::slotHotcuePaletteChanged(const QString& paletteName) {
+    QPixmap preview = drawPalettePreview(paletteName);
+    labelHotcuePalette->setPixmap(preview);
+
+    ColorPalette palette = mixxx::PredefinedColorPalettes::kDefaultTrackColorPalette;
+    foreach (const ColorPalette& pal, mixxx::PredefinedColorPalettes::kPalettes) {
```

Comment:
Why use Qt `foreach` macro instead of C++ standard `for (x : y)`?

Reply:
This loop can also be replaced with ColorPaletteSettings.getPalette I think